File: app/internal/publisher/template/c/device_replace.c
#include "device_replace.h"
#include <stdio.h>
#include <string.h>

JsonMap* wsd_device_replace_show(LicenseEngine* engine, const char* license_key) {
    (void)license_key;

    const char* support_email = "support@websmithdigital.com";
    if (engine && engine->config) {
        const char* branding_str = wsd_json_get(engine->config, "branding_json");
        if (branding_str) {
            JsonMap* branding = wsd_json_parse(branding_str);
            if (branding) {
                const char* email = wsd_json_get(branding, "support_email");
                if (email) support_email = email;
                wsd_json_free(branding);
            }
        }
    }

    printf("========================================\n");
    printf("  Replace Device\n");
    printf("========================================\n");
    printf("\n");
    printf("  Device reactivation requires Websmith Support approval.\n");
    printf("\n");
    printf("  Please contact support at: %s\n", support_email);
    printf("  The application will remain locked until reactivation is approved.\n");
    printf("\n");

    JsonMap* result = wsd_json_map_new();
    wsd_json_set(result, "action", "contact_support");
    wsd_json_set(result, "support_email", support_email);
    return result;
}

File: app/internal/publisher/template/typescript/device_replace.ts
import * as readline from 'readline';
import { LicenseEngine } from './license_engine';

export class DeviceReplaceDialog {
  private engine: LicenseEngine;
  private config: Record<string, any>;
  private licenseKey: string;
  result: Record<string, any> | null = null;
  private supportEmail: string;

  constructor(engine: LicenseEngine, licenseKey: string) {
    this.engine = engine;
    this.config = engine.config;
    this.licenseKey = licenseKey;
    engine._licenseKey = licenseKey;
    this.supportEmail = this.config.branding?.support_email || 'support@websmithdigital.com';
  }

  async show(): Promise<Record<string, any> | null> {
    const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
    const question = (q: string): Promise<string> => new Promise((resolve) => rl.question(q, resolve));

    console.log('=== Reactivate License ===');
    console.log('Device reactivation requires Websmith Support approval.');
    console.log(`Please contact ${this.supportEmail} to submit a reactivation request.`);
    console.log('');
    console.log('Your application will remain locked until the reactivation is approved.');
    console.log('');

    await question('Press Enter to return...');
    this.result = { action: 'contact_support', support_email: this.supportEmail };
    rl.close();
    return this.result;
  }
}
